### ISSUE TITLE
No application found error from `okta_app` data source

### ISSUE THREAD

**Issue by PurrBiscuit** [okta/terraform-provider-okta#1111]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform version: 1.0.11
Okta provider version: 3.26.0

### Affected Resource(s)

- `data.okta_app`

### Terraform Configuration Files

```hcl
data "okta_app" "service" {
  label = "360-subtask-frontend"
}
```

### Debug Output

Here's a truncated debug log that shows the two api calls from that data source to look for the app - https://gist.github.com/PurrBiscuit/4cadd59632c3d79c72de72ba9e73820e.  I didn't want to provide the full log since there were a bunch of secret values in there for other resources.

### Expected Behavior

We have two apps in our Okta account that both begin with `360-subtask-frontend`...one is just `360-subtask-frontend` and the other is `360-subtask-frontend-dev`.  Using the `data.okta_app` data source with the label set to "360-subtask-frontend", we'd expect that the `360-subtask-frontend` named app would be returned.

### Actual Behavior

We're receiving the following error during the `terraform plan` stage:

```
│ Error: no application found with provided label: 360-subtask-frontend
│
│   with data.okta_app.service,
│   on data_sources.tf line 1, in data "okta_app" "service":
│    1: data "okta_app" "service" {
```

Side note - if we rename the `360-subtask-frontend-dev` to `dev-360-subtask-frontend` in our Okta account then the data source listed above with the "360-subtask-frontend" label runs fine and returns the correct app.

### Steps to Reproduce

1. Create an app named `360-subtask-frontend-dev` in an Okta account.
2. Create another app named `360-subtask-frontend` in the same Okta account.
3. Create a Terraform configuration to search for the `360-subtask-frontend` app:
```hcl
data "okta_app" "service" {
  label = "360-subtask-frontend"
}
```
4. Receive the `no application found with provided label` error from the provider.

### Important Factoids

The app data source is using the `q` parameter to filter apps by `label` using the Okta Apps API.  According to that API's documentation for the List Apps request, the [`q` parameter doesn't check for equality, but rather the `startsWith` operator](https://developer.okta.com/docs/reference/api/apps/#list-applications).  It looks like `label` and `label_prefix` are essentially checking for the same thing for this data source.  From the debug logging, I can see that the `360-subtask-frontend` app does get returned in the second request, after the `next` link is found, but for some reason the provider is throwing an error, saying that the app couldn't be found at all.

**Comment by monde** [comment]
Released as [v3.28.0](https://github.com/okta/terraform-provider-okta/releases/tag/v3.28.0).

**Comment by PurrBiscuit** [comment]
Thanks for the quick turn around on this one!
